addons/website_crm_partner_assign/models/crm_lead.py

```diff
@@ -30,6 +30,10 @@ def _merge_data(self, fields):
         fields += ['partner_latitude', 'partner_longitude', 'partner_assigned_id', 'date_partner_assign']
         return super(CrmLead, self)._merge_data(fields)
 
+    def edit_lead_contact_portal(self, values):
```

Comment:
Maybe ``update_contact_details_from_portal```? And whitelist what we give to values, to avoid side-effects like allowing people to send any arbitrary values to all fields (using sudo may allow people to update 2many fields they would not be able to do under normal write grant access).